e2e_tests/tests/test_sdk.py

```diff
@@ -83,6 +86,98 @@ def test_completed_experiment_and_checkpoint_apis(client: _client.Determined) ->
     assert "newkey" not in client.get_checkpoint(ckpt.uuid).metadata
 
 
+@pytest.mark.e2e_cpu
+def test_checkpoint_apis(client: _client.Determined) -> None:
+    with open(conf.fixtures_path("no_op/single-default-ckpt.yaml")) as f:
+        config = yaml.safe_load(f)
+
+    # Test for 100 batches/checkpoint every 10 = 10 checkpoints.
+    config["min_checkpoint_period"]["batches"] = 10
+    config["min_validation_period"]["batches"] = 10
+    config["checkpoint_storage"] = {}
+    config["checkpoint_storage"]["save_trial_best"] = 10
+
+    exp = client.create_experiment(config, conf.fixtures_path("no_op"))
+
+    # Await first trial is safe to call before a trial has started.
+    trial = exp.await_first_trial()
+
+    assert exp.wait() == _client.ExperimentState.COMPLETED
+    trials = exp.get_trials()
+    assert len(trials) == 1, trials
+
+    checkpoints = trial.get_checkpoints()
+    assert len(checkpoints) == 10
+
+    # Validate end (report) time sorting.
+    checkpoints = trial.get_checkpoints(
+        sort_by=_client.CheckpointSortBy.END_TIME, order_by=_client.CheckpointOrderBy.DESC
+    )
+    end_times = [checkpoint.report_time for checkpoint in checkpoints]
+    assert all(x >= y for x, y in zip(end_times, end_times[1:]))
+
+    # Validate state sorting.
+    checkpoints = trial.get_checkpoints(
+        sort_by=_client.CheckpointSortBy.STATE, order_by=_client.CheckpointOrderBy.ASC
+    )
+    states = [checkpoint.state.value for checkpoint in checkpoints]
+    assert all(x <= y for x, y in zip(states, states[1:]))
+
+    # Validate UUID sorting.
+    checkpoints = trial.get_checkpoints(
+        sort_by=_client.CheckpointSortBy.UUID, order_by=_client.CheckpointOrderBy.ASC
+    )
+    uuids = [checkpoint.uuid for checkpoint in checkpoints]
+    assert all(x <= y for x, y in zip(uuids, uuids[1:]))
+
+    # Validate batch number sorting.
+    checkpoints = trial.get_checkpoints(
+        sort_by=_client.CheckpointSortBy.BATCH_NUMBER, order_by=_client.CheckpointOrderBy.DESC
+    )
+    batch_numbers = [checkpoint.metadata["steps_completed"] for checkpoint in checkpoints]
+    assert all(x >= y for x, y in zip(batch_numbers, batch_numbers[1:]))
+
+    # Validate metric sorting.
+    checkpoints = trial.get_checkpoints(
+        sort_by="validation_error", order_by=_client.CheckpointOrderBy.ASC
+    )
+    validation_metrics = [
+        checkpoint.training.validation_metrics["avgMetrics"]["validation_error"]
+        for checkpoint in checkpoints
+    ]
+    assert all(x <= y for x, y in zip(validation_metrics, validation_metrics[1:]))
+
+    # Expect 10 completed checkpoints.
+    checkpoints = [
+        checkpoint
+        for checkpoint in checkpoints
+        if checkpoint.state == _client.CheckpointState.COMPLETED
+    ]
+    assert len(checkpoints) == 10
+
+    # Delete first checkpoint.
```

Comment:
Tests looking good to me.

Should we also check that calling delete on the same checkpoint twice doesn't error? I recall some discussion/worry around this in a conversation with @rb-determined-ai

Reply:
there's [a separate PR](https://github.com/determined-ai/determined/pull/6247) for handling this in the shared_fs use case.